File: app/public/wp-content/themes/biber-helmets/crop_assets.py
"""
Script cắt tự động 10 ảnh bài viết từ Ảnh 1 và 8 ảnh mũ bảo hiểm từ Ảnh 2
Dùng thư viện Pillow (PIL)
"""
import os
from PIL import Image

BRAIN_DIR = r"C:\Users\DELL\.gemini\antigravity-ide\brain\5856ad80-eeba-4418-8864-3975bbbf4396\.user_uploaded"
THEME_DIR = r"E:\biker so 1\team biker\app\public\wp-content\themes\biber-helmets"
IMG_DIR = os.path.join(THEME_DIR, "assets", "images")

os.makedirs(IMG_DIR, exist_ok=True)

# 1. Logo
logo_src = os.path.join(BRAIN_DIR, "media_1789593552551.png")
logo_dst = os.path.join(IMG_DIR, "biker-logo.png")
if os.path.exists(logo_src):
    try:
        im = Image.open(logo_src)
        im.save(logo_dst)
        print(f"[OK] Logo saved to {logo_dst}")
    except Exception as e:
        print(f"[ERR] Logo error: {e}")

# 2. Cắt 10 ảnh bài viết từ Ảnh 1 (media_1789593307046.jpg)
news_src = os.path.join(BRAIN_DIR, "media_1789593307046.jpg")
if not os.path.exists(news_src):
    news_src = os.path.join(IMG_DIR, "news-grid-full.jpg")

if os.path.exists(news_src):
    try:
        im = Image.open(news_src)
        w, h = im.size
        # Copy full grid
        im.save(os.path.join(IMG_DIR, "news-grid-full.jpg"), quality=90)
        
        boxes = {
            1:  (int(w * 0.16), int(h * 0.02), int(w * 0.32), int(h * 0.35)),
            2:  (int(w * 0.33), int(h * 0.02), int(w * 0.49), int(h * 0.28)),
            3:  (int(w * 0.50), int(h * 0.02), int(w * 0.66), int(h * 0.42)),
            4:  (int(w * 0.67), int(h * 0.02), int(w * 0.83), int(h * 0.28)),
            5:  (int(w * 0.84), int(h * 0.02), int(w * 0.99), int(h * 0.30)),
            6:  (int(w * 0.33), int(h * 0.30), int(w * 0.49), int(h * 0.52)),
            7:  (int(w * 0.16), int(h * 0.45), int(w * 0.32), int(h * 0.66)),
            8:  (int(w * 0.33), int(h * 0.53), int(w * 0.49), int(h * 0.85)),
            9:  (int(w * 0.84), int(h * 0.32), int(w * 0.99), int(h * 0.56)),
            10: (int(w * 0.67), int(h * 0.65), int(w * 0.83), int(h * 0.95)),
        }
        for i, box in boxes.items():
            crop_im = im.crop(box)
            crop_path = os.path.join(IMG_DIR, f"news-{i}.jpg")
            crop_im.save(crop_path, quality=90)
            print(f"[OK] News {i} saved -> {crop_path}")
    except Exception as e:
        print(f"[ERR] News crop error: {e}")

# 3. Cắt 8 ảnh mũ bảo hiểm từ Ảnh 2 (media_1789593488232.png)
helm_src = os.path.join(BRAIN_DIR, "media_1789593488232.png")
if not os.path.exists(helm_src):
    helm_src = os.path.join(IMG_DIR, "helmets-grid-full.png")

if os.path.exists(helm_src):
    try:
        im = Image.open(helm_src)
        w, h = im.size
        im.save(os.path.join(IMG_DIR, "helmets-grid-full.png"))
        
        boxes2 = {
            1: (int(w * 0.70), int(h * 0.10), int(w * 0.84), int(h * 0.43)),
            2: (int(w * 0.27), int(h * 0.11), int(w * 0.41), int(h * 0.45)),
            3: (int(w * 0.41), int(h * 0.00), int(w * 0.55), int(h * 0.27)),
            4: (int(w * 0.27), int(h * 0.49), int(w * 0.41), int(h * 0.78)),
            5: (int(w * 0.56), int(h * 0.53), int(w * 0.70), int(h * 0.84)),
            6: (int(w * 0.12), int(h * 0.21), int(w * 0.26), int(h * 0.46)),
            7: (int(w * 0.41), int(h * 0.58), int(w * 0.55), int(h * 0.85)),
            8: (int(w * 0.70), int(h * 0.52), int(w * 0.84), int(h * 0.82)),
        }
        for i, box in boxes2.items():
            crop_im = im.crop(box)
            crop_path = os.path.join(IMG_DIR, f"helmet-{i}.jpg")
            crop_im.convert("RGB").save(crop_path, quality=90)
            print(f"[OK] Helmet {i} saved -> {crop_path}")
    except Exception as e:
        print(f"[ERR] Helmet crop error: {e}")

print("Done all asset cropping!")

File: app/public/wp-content/themes/biber-helmets/crop_assets.php
<?php
/**
 * Tự động trích xuất 10 ảnh bài viết từ Ảnh 1, 8 ảnh sản phẩm từ Ảnh 2, và Logo từ Ảnh 3
 */
function biker_extract_all_media_assets() {
    $brain_dir = 'C:/Users/DELL/.gemini/antigravity-ide/brain/5856ad80-eeba-4418-8864-3975bbbf4396/.user_uploaded/';
    $img_dir   = get_template_directory() . '/assets/images/';
    
    if (!is_dir($img_dir)) {
        @mkdir($img_dir, 0777, true);
    }

    // 1. Sao chép Logo (Ảnh 3)
    $logo_src = file_exists($brain_dir . 'media_1789593552551.png') ? $brain_dir . 'media_1789593552551.png' : $img_dir . 'biker-logo.png';
    $logo_dst = $img_dir . 'biker-logo.png';
    if (file_exists($logo_src) && $logo_src !== $logo_dst && (!file_exists($logo_dst) || filesize($logo_dst) < 1000)) {
        @copy($logo_src, $logo_dst);
    }

    // 2. Cắt 10 ảnh bài viết từ Ảnh 1 (media_1789593307046.jpg)
    $news_src = file_exists($brain_dir . 'media_1789593307046.jpg') ? $brain_dir . 'media_1789593307046.jpg' : $img_dir . 'news-grid-full.jpg';
    if (file_exists($news_src) && function_exists('imagecreatefromjpeg') && !file_exists($img_dir . 'news-10.jpg')) {
        $src_img = @imagecreatefromjpeg($news_src);
        if ($src_img) {
            $w = imagesx($src_img);
            $h = imagesy($src_img);
            if (!file_exists($img_dir . 'news-grid-full.jpg') && $news_src !== $img_dir . 'news-grid-full.jpg') {
                @copy($news_src, $img_dir . 'news-grid-full.jpg');
            }

            $boxes = array(
                1  => array((int)($w * 0.16), (int)($h * 0.02), (int)($w * 0.16), (int)($h * 0.33)),
                2  => array((int)($w * 0.33), (int)($h * 0.02), (int)($w * 0.16), (int)($h * 0.26)),
                3  => array((int)($w * 0.50), (int)($h * 0.02), (int)($w * 0.16), (int)($h * 0.40)),
                4  => array((int)($w * 0.67), (int)($h * 0.02), (int)($w * 0.16), (int)($h * 0.26)),
                5  => array((int)($w * 0.84), (int)($h * 0.02), (int)($w * 0.15), (int)($h * 0.28)),
                6  => array((int)($w * 0.33), (int)($h * 0.30), (int)($w * 0.16), (int)($h * 0.22)),
                7  => array((int)($w * 0.16), (int)($h * 0.45), (int)($w * 0.16), (int)($h * 0.21)),
                8  => array((int)($w * 0.33), (int)($h * 0.53), (int)($w * 0.16), (int)($h * 0.32)),
                9  => array((int)($w * 0.84), (int)($h * 0.32), (int)($w * 0.15), (int)($h * 0.24)),
                10 => array((int)($w * 0.67), (int)($h * 0.65), (int)($w * 0.16), (int)($h * 0.30)),
            );

            foreach ($boxes as $i => $b) {
                $crop = @imagecreatetruecolor($b[2], $b[3]);
                if ($crop) {
                    @imagecopy($crop, $src_img, 0, 0, $b[0], $b[1], $b[2], $b[3]);
                    @imagejpeg($crop, $img_dir . "news-{$i}.jpg", 90);
                    @imagedestroy($crop);
                }
            }
            @imagedestroy($src_img);
        }
    }

    // 3. Cắt 8 ảnh mũ bảo hiểm từ Ảnh 2 (media_1789593488232.png)
    $helmets_src = file_exists($brain_dir . 'media_1789593488232.png') ? $brain_dir . 'media_1789593488232.png' : $img_dir . 'helmets-grid-full.png';
    if (file_exists($helmets_src) && function_exists('imagecreatefrompng') && !file_exists($img_dir . 'helmet-8.jpg')) {
        $src_img = @imagecreatefrompng($helmets_src);
        if ($src_img) {
            $w = imagesx($src_img);
            $h = imagesy($src_img);
            if (!file_exists($img_dir . 'helmets-grid-full.png') && $helmets_src !== $img_dir . 'helmets-grid-full.png') {
                @copy($helmets_src, $img_dir . 'helmets-grid-full.png');
            }

            $boxes2 = array(
                1 => array((int)($w * 0.70), (int)($h * 0.10), (int)($w * 0.14), (int)($h * 0.33)),
                2 => array((int)($w * 0.27), (int)($h * 0.11), (int)($w * 0.14), (int)($h * 0.34)),
                3 => array((int)($w * 0.41), (int)($h * 0.00), (int)($w * 0.14), (int)($h * 0.27)),
                4 => array((int)($w * 0.27), (int)($h * 0.49), (int)($w * 0.14), (int)($h * 0.29)),
                5 => array((int)($w * 0.56), (int)($h * 0.53), (int)($w * 0.14), (int)($h * 0.31)),
                6 => array((int)($w * 0.12), (int)($h * 0.21), (int)($w * 0.14), (int)($h * 0.25)),
                7 => array((int)($w * 0.41), (int)($h * 0.58), (int)($w * 0.14), (int)($h * 0.27)),
                8 => array((int)($w * 0.70), (int)($h * 0.52), (int)($w * 0.14), (int)($h * 0.30)),
            );

            foreach ($boxes2 as $i => $b) {
                $crop = @imagecreatetruecolor($b[2], $b[3]);
                if ($crop) {
                    @imagecopy($crop, $src_img, 0, 0, $b[0], $b[1], $b[2], $b[3]);
                    @imagejpeg($crop, $img_dir . "helmet-{$i}.jpg", 90);
                    @imagedestroy($crop);
                }
            }
            @imagedestroy($src_img);
        }
    }
}
